test/framework/src/main/java/org/elasticsearch/test/VersionUtils.java

```diff
@@ -248,4 +248,18 @@ public static Version maxCompatibleVersion(Version version) {
         assert compatible.size() > 0;
         return compatible.get(compatible.size() - 1);
     }
+
+    /**
+     * Returns a random version index compatible with the current version.
+     */
+    public static Version randomIndexCompatibleVersion(Random random) {
+        return randomVersionBetween(random, Version.CURRENT.minimumIndexCompatibilityVersion(), Version.CURRENT);
+    }
+
+    /**
+     * Returns a random version wire compatible with the current version.
+     */
+    public static Version randomWireCompatibleVersion(Random random) {
```

Comment:
nothing seems to be using this. do you see concrete situations where this will be used in follow-up PRs related to #41164?

Reply:
On further look, it seems we actually already have something for this. The naming is confusing because it is just "compatible version", see `VersionUtils.randomCompatibleVersion` and `Version.isCompatible`. I will remove this method from here and do a follow up to rename that for clarity compared to index compatibility.

Reply:
oh. good catch! I didn't notice that either. follow-up makes sense 👍